src/nfa/nfa.cc

```diff
@@ -312,6 +312,136 @@ struct StackLevel {
     };
 };
 
+struct NodeData {
```

Comment:
This should be kept in an anonymous namespace because users are not supposed to use this structure manually.